ttnn/cpp/ttnn/operations/conv/conv2d/device/conv2d_op.hpp

```diff
@@ -19,7 +19,7 @@ namespace conv2d {
 constexpr uint32_t l1_scratchpad_CB_size = 64;
 struct Conv2dConfig {
     tt::tt_metal::DataType dtype = tt::tt_metal::DataType::BFLOAT16;
-    tt::tt_metal::DataType weights_dtype = tt::tt_metal::DataType::BFLOAT16;
+    std::optional<tt::tt_metal::DataType> weights_dtype = std::nullopt;
```

Comment:
Pls add a comment above this property saying when it should be set and what happens if it's not set.